---
title: "Parallel処理の後にActiverecordで検索しようとすると通信エラーが発生する"
draft: false
tags: ["ruby","activerecord"]
private: false
slug: "ddddd4c5-0d7e-462f-9d64-d8096812f488"
date: "2019-01-30T12:25:19+09:00"
lastmod: "2019-01-30T12:26:05+09:00"
keywords: ["ruby","activerecord","ベジプロ","プログ","プログラム"]
# headless: true
---

# 対応
検索を行う前に、コネクションを再接続する
```rb
SomeModel.connection.reconnect!
SomeModel.find(1)
```

# 事象
## メッセージ
`ActiveRecord::StatementInvalid: Mysql2::Error: MySQL server has gone away: SELECT...`

## 処理
```rb
Parallel.map(SomeModel.all, in_process: 4) do |some_model|
  # something to do...
end

SomeModel.find(1) # <- ここで発生
```
